VirtualFleetDocker.py

```diff
@@ -0,0 +1,257 @@
+TMP_CONFIG_PATH = os.path.join(CONFIG_PATH, "tmp-configs")
+EXTERNAL_SERVER_CONFIG = os.path.join(CONFIG_PATH, "external-server/config.json")
+MODULE_GATEWAY_CONFIG = os.path.join(CONFIG_PATH, "module-gateway/config.json")
+VERNEMQ_CONFIG = os.path.join(CONFIG_PATH, "vernemq")
+LOGS_DIR = "logs"
+runningContainers = []
+
+class FileDoesntExistException(Exception):
+    pass
+
+class PortOutOfRangeException(Exception):
+    def __init__(self, message="Ports out of range, it has to be: 1024 < my port < 65535"):
+        super().__init__(message)
+
+def parse_arguments():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--config", default=os.path.join(CONFIG_PATH, "virtual-fleet-config.json"), help="path to json configuration file")
+    return parser.parse_args()
+
+def check_paths(arguments):
+    if not os.path.exists(arguments.config):
+        raise FileDoesntExistException(f"Json file {arguments.config} doesn't exist")
+
+def is_port_available(port):
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        return s.connect_ex(("localhost", port)) != 0
+
+def check_vehicles_uniqueness(vehicles):
+    vehicle_names = set()
+    for vehicle in vehicles:
+        vehicle_id = f"{vehicle['company']}-{vehicle['name']}"
+        if vehicle_id in vehicle_names:
+            raise ValueError(f"Vehicle {vehicle_id} is not unique")
+        vehicle_names.add(vehicle_id)
+
+def create_config_files(vehicle):
+    with open(os.path.abspath(EXTERNAL_SERVER_CONFIG), "r") as file:
+        config = json.load(file)
+    config["company_name"] = vehicle["company"]
+    config["car_name"] = vehicle["name"]
+    tmp_config_address = os.path.join(TMP_CONFIG_PATH, f"{vehicle['company']}-{vehicle['name']}-external-server-config.json")
+    with open(os.path.abspath(tmp_config_address), "w") as file:
+        json.dump(config, file, indent=4)
+
+    with open(os.path.abspath(MODULE_GATEWAY_CONFIG), "r") as file:
+        config = json.load(file)
+    config["external-connection"]["company"] = vehicle["company"]
+    config["external-connection"]["vehicle-name"] = vehicle["name"]
+    tmp_config_address = os.path.join(TMP_CONFIG_PATH, f"{vehicle['company']}-{vehicle['name']}-gateway-config.json")
+    with open(os.path.abspath(tmp_config_address), "w") as file:
+        json.dump(config, file, indent=4)
+
+def is_container_running(docker_client, image_name):
+    for container in docker_client.containers.list():
+        if image_name in container.image.tags:
+            return container
+    return None
+
+def run_program(arguments):
+    with open(arguments.config) as file:
+        settings = json.load(file)
+
+    check_vehicles_uniqueness(settings["vehicles"])
+
+    client = docker.from_env()
+    port = settings["start-port"]
+    logging.info("Starting docker containers")
+
+    start_mqtt_broker_container(client, settings)
+
+    os.makedirs(os.path.abspath(TMP_CONFIG_PATH), mode=0o777, exist_ok=True)
```

Comment:
Is it possible to easily create config files before? In case you want to update something about it, it will be quite hard to find where what is happening -> move create_config_files and this line together before run_program